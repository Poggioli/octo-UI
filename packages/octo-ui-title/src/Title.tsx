import { CSS, forwardRef } from "@octo-ui/core";
import { StyledTitle } from "./styles";
import { DEFAULT_TAG_TITLE, TitleProps } from "./types";

const Title = forwardRef<typeof StyledTitle, TitleProps>(
  ({ color, ...props }, forwardedRef) => {
    const css: CSS = {
      color: color ?? "$grey12",
      ...props.css,
    };

    return (
      <StyledTitle
        as={DEFAULT_TAG_TITLE}
        {...props}
        ref={forwardedRef}
        size={{
          "@initial": "8",
          "@medium": "9",
        }}
        css={{
          fontWeight: 500,
          fontVariantNumeric: "proportional-nums",
          lineHeight: "35px",
          ...css,

          "@medium": {
            lineHeight: "55px",
            ...css["@medium"],
          },
        }}
      />
    );
  }
);
Title.toString = () => `.${StyledTitle.className}`;

export { StyledTitle };
export { Title };
