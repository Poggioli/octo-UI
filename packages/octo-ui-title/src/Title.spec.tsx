import { render } from "@testing-library/react";
import { Title } from "./Title";

describe("<Title />", () => {
  it(`GIVEN a <Title />
      WHEN screen is small than 900px
      THEN should render a <Title /> using size 8
      AND lineHeight 35px`, () => {
    global.innerWidth = 768;
    const rendered = render(<Title>Title</Title>);
    expect(rendered.container).toMatchSnapshot();
  });

  it(`GIVEN a <Title />
      WHEN screen is equal or greater than 900px
      THEN should render a <Title /> using size 9
      AND lineHeight 55px`, () => {
    global.innerWidth = 900;
    const rendered = render(<Title>Title</Title>);
    expect(rendered.container).toMatchSnapshot();
  });

  it(`GIVEN a <Title />
      WHEN toString is called
      THEN return a string`, () => {
    expect(Title.toString()).toBe(".octo-UI-c-gWerhV");
  });

  describe("color props", () => {
    it(`GIVEN a <Title />
        WHEN color props is not setted
        THEN should render a <Title /> $grey12 as color`, () => {
      const rendered = render(<Title>Title</Title>);
      expect(rendered.container).toMatchSnapshot();
    });

    it(`GIVEN a <Title />
        WHEN color props is $grey11
        THEN should render a <Title /> $grey11 as color`, () => {
      const rendered = render(<Title color="$grey11">Title</Title>);
      expect(rendered.container).toMatchSnapshot();
    });
  });
});
