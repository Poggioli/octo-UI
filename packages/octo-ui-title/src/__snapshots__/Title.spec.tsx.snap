// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Title /> GIVEN a <Title />
      WHEN screen is equal or greater than 900px
      THEN should render a <Title /> using size 9
      AND lineHeight 55px 1`] = `
<div>
  <h1
    class="octo-UI-c-gWerhV octo-UI-c-PJLV octo-UI-c-gWerhV-cpZglh-size-8 octo-UI-c-gWerhV-CWrEW-size-9 octo-UI-c-gWerhV-ikUlEhX-css"
  >
    Title
  </h1>
</div>
`;

exports[`<Title /> GIVEN a <Title />
      WHEN screen is small than 900px
      THEN should render a <Title /> using size 8
      AND lineHeight 35px 1`] = `
<div>
  <h1
    class="octo-UI-c-gWerhV octo-UI-c-PJLV octo-UI-c-gWerhV-cpZglh-size-8 octo-UI-c-gWerhV-CWrEW-size-9 octo-UI-c-gWerhV-ikUlEhX-css"
  >
    Title
  </h1>
</div>
`;

exports[`<Title /> color props GIVEN a <Title />
        WHEN color props is $grey11
        THEN should render a <Title /> $grey11 as color 1`] = `
<div>
  <h1
    class="octo-UI-c-gWerhV octo-UI-c-PJLV octo-UI-c-gWerhV-cpZglh-size-8 octo-UI-c-gWerhV-CWrEW-size-9 octo-UI-c-gWerhV-ijOTmEM-css"
  >
    Title
  </h1>
</div>
`;

exports[`<Title /> color props GIVEN a <Title />
        WHEN color props is not setted
        THEN should render a <Title /> $grey12 as color 1`] = `
<div>
  <h1
    class="octo-UI-c-gWerhV octo-UI-c-PJLV octo-UI-c-gWerhV-cpZglh-size-8 octo-UI-c-gWerhV-CWrEW-size-9 octo-UI-c-gWerhV-ikUlEhX-css"
  >
    Title
  </h1>
</div>
`;
